runtime/interpreter/interpreter.go

```diff
@@ -2214,8 +2217,182 @@ func (interpreter *Interpreter) writeStored(
+	return -1 < val.Cmp(low) && val.Cmp(high) < 1
+}
+
+var FromStringFunctionValues = func() map[string]FromStringFunctionValue {
+	declarations := []FromStringFunctionValue{
+		fromStringDeclaration(sema.Int8Type, fromStringImplSigned(8, func(n int64) Value {
+			return NewUnmeteredInt8Value(int8(n))
+		})),
+		fromStringDeclaration(sema.Int16Type, fromStringImplSigned(16, func(n int64) Value {
+			return NewUnmeteredInt16Value(int16(n))
+		})),
+		fromStringDeclaration(sema.Int32Type, fromStringImplSigned(32, func(n int64) Value {
+			return NewUnmeteredInt32Value(int32(n))
+		})),
+		fromStringDeclaration(sema.Int64Type, fromStringImplSigned(64, func(n int64) Value {
+			return NewUnmeteredInt64Value(n)
+		})),
+		fromStringDeclaration(sema.Int128Type, fromStringImplBig(func(b *big.Int) (Value, bool) {
+			ok := inRange(b, sema.Int128TypeMinIntBig, sema.Int128TypeMaxIntBig)
+			return Int128Value{b}, ok
+		})),
+		fromStringDeclaration(sema.Int256Type, fromStringImplBig(func(b *big.Int) (Value, bool) {
+			ok := inRange(b, sema.Int256TypeMinIntBig, sema.Int256TypeMaxIntBig)
+			return Int256Value{b}, ok
+		})),
+		fromStringDeclaration(sema.IntType, fromStringImplBig(func(b *big.Int) (Value, bool) {
+			return IntValue{b}, true
+		})),
+		fromStringDeclaration(sema.UInt8Type, fromStringImplUnsigned(8, func(n uint64) Value {
+			return NewUnmeteredUInt8Value(uint8(n))
+		})),
+		fromStringDeclaration(sema.UInt16Type, fromStringImplUnsigned(16, func(n uint64) Value {
+			return NewUnmeteredUInt16Value(uint16(n))
+		})),
+		fromStringDeclaration(sema.UInt32Type, fromStringImplUnsigned(32, func(n uint64) Value {
+			return NewUnmeteredUInt32Value(uint32(n))
+		})),
+		fromStringDeclaration(sema.UInt64Type, fromStringImplUnsigned(64, func(n uint64) Value {
+			return NewUnmeteredUInt64Value(n)
+		})),
+		fromStringDeclaration(sema.UInt128Type, fromStringImplBig(func(b *big.Int) (Value, bool) {
+			ok := inRange(b, sema.UInt128TypeMinIntBig, sema.UInt128TypeMaxIntBig)
+			return UInt128Value{b}, ok
+		})),
+		fromStringDeclaration(sema.UInt256Type, fromStringImplBig(func(b *big.Int) (Value, bool) {
+			ok := inRange(b, sema.UInt256TypeMinIntBig, sema.UInt256TypeMaxIntBig)
+			return UInt256Value{b}, ok
+		})),
+		fromStringDeclaration(sema.UIntType, fromStringImplBig(func(b *big.Int) (Value, bool) {
+			return UIntValue{b}, true
+		})),
+		fromStringDeclaration(sema.Word8Type, fromStringImplUnsigned(8, func(n uint64) Value {
+			return NewUnmeteredWord8Value(uint8(n))
+		})),
+		fromStringDeclaration(sema.Word16Type, fromStringImplUnsigned(16, func(n uint64) Value {
+			return NewUnmeteredWord16Value(uint16(n))
+		})),
+		fromStringDeclaration(sema.Word32Type, fromStringImplUnsigned(32, func(n uint64) Value {
+			return NewUnmeteredWord32Value(uint32(n))
+		})),
+		fromStringDeclaration(sema.Word64Type, fromStringImplUnsigned(64, func(n uint64) Value {
+			return NewUnmeteredWord64Value(n)
```

Comment:
The return type for the `fromString()` methods is always `OptionalType`. Therefore, all these methods should return an `OptionalValue` (i.e: wrap with `SomeValue`)

Reply:
they're wrapped in the `fromStringImpl-` series of functions, these are just converters to avoid duplicated code

Reply:
ah right, didn't notice that one 👍